text/1003-deprecation-import-ember-from-ember.md

```diff
@@ -0,0 +1,344 @@
+
+Many of these APIs already have alternatives, and those will be called out explicitly in the _Transition Path_ below.
+
+## Transition Path
+
+This list is semi-exhaustive, in that it covers _every_ export from 'ember', but may not exhaustivily provide alternatives.
+
+Throughout the rest of this RFC, the following key will be used:
+- 🌐 to mean "this is public API"
+- 🔒 to mean "this is private API"
+- 🧷 to mean "this is protected API"
+- 🫣 to mean "no declared access"
+
+### Testing utilities
+
+APIs for wiring up a test framework (e.g. QUnit, _etc_)
+- 🌐 `Ember.Test`
+- 🌐 `Ember.Test.Adapter` - currently available at [`@ember/test`](https://api.emberjs.com/ember/5.6/modules/@ember%2Ftest)
+- 🌐 `Ember.Test.QUnitAdapter`
+- 🌐 `Ember.setupForTesting`
+
+These will need to be moved to a module such as `@ember/test`.
+
+### A way to communicate with the ember-inspector
+
+The inspector will be hit especially hard by the removal of these APIs.
+
+A good few already have available imports though.
+
+|   | API | import |
+| - | --- | ------ |
+|🔒| `Ember.meta` | `import { meta } from '@ember/-internals/meta';` |
+|🌐| `Ember.VERSION` | none |
+|🔒| `Ember._captureRenderTree` | `import { captureRenderTree } from '@ember/debug';` |
+|🔒| `Ember.instrument` | `import { instrument } from '@ember/instrumentation';` |
+|🔒| `Ember.subscribe` | `import { subscribe } from '@ember/instrumentation';` |
+|🔒| `Ember.Instrumentation.*` | `import { * } from '@ember/instrumentation';`  |
+|🫣| `Ember.ViewUtils` | `import * as viewUtils from '@ember/-internals/views';`[^view-utils]  |
+|🔒| `Ember.ViewUtils.getChildViews` | `import { getChildViews } from '@ember/-internals/views';` |
+|🫣| `Ember.ViewUtils.getElementView` | `import { getElementView } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getRootViews` | `import { getRootViews } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getViewBounds` | `import { getViewBounds } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getViewBoundingClientRect` | `import { getViewBoundingClientRect } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getViewClientRects` | `import { getViewClientRects } from '@ember/-internals/views';` |
+|🔒| `Ember.ViewUtils.getViewElement`  | `import { getViewElement } from '@ember/-internals/views';` |
+|🫣| `Ember.ViewUtils.isSimpleClick` | `import { isSimpleClick } from '@ember/-internals/views';` |
+|🫣| `Ember.ViewUtils.isSerializationFirstNode` | `import { isSerializationFirstNode } from '@ember/-internals/glimmer';` |
+
+[^view-utils]: Not all of these exports are used for `ViewUtils`.
+
+
+Perhaps we can have folks add this to their apps:
+```js
+import { macroCondition, isDevelopingApp, importSync } from '@embroider/macros';
+
+if (macroCondition(isDevelopingApp())) {
+  // maybe this is side-effecting and installs 
+  // some functions on `globalThis` that the inspector could call
+  importSync('@ember/inspector-support');
+}
```

Comment:
Using Ember Inspector on production build helps with debugging from time to time. What is the trade-off of supporting Ember Inspector in production build as well?

Reply:
My personal preference would be to have folks opt in to that by removing the condition and just importing inspector-support